galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.ceda.manager.ivt/src/main/java/dev/galasa/cicsts/ceda/manager/ivt/CedaManagerIVT.java

```diff
@@ -203,49 +157,90 @@ public void testResourceLibrary() throws TextNotFoundException, ErrorTextFoundEx
-      cics.ceda().createResource(cedaTerminal, resourceType, resourceName3, groupName, null);
-      /** Different group **/
-      cics.ceda().createResource(cedaTerminal, resourceType, resourceName4, groupName2, null);
-      // Installing only one group and check if installed group appeared in CEMT and not installed one did not
-      cics.ceda().installGroup(cedaTerminal, groupName);
-      if (cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName).containsValue(resourceName.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName2).containsValue(resourceName2.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName3).containsValue(resourceName3.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName4)==null) {
-         result =true;
-      }
-      assertThat(result).isEqualTo(true);
-
-      // Checking if group delete works by discarding elements from CEMT and deleting group from CEDA, checking by installing group
-      if (result) {
-         result=false;
-        cics.ceda().deleteGroup(cedaTerminal, groupName);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName2);
-        cics.cemt().discardResource(cemtTerminal, resourceType, resourceName3);
-        assertThatThrownBy(() -> {
-           cics.ceda().installGroup(cedaTerminal, groupName);
-        }).isInstanceOf(CedaException.class).hasMessageContaining("Problem determining the result from the CEDA command");
-
-      } else Fail.fail("CEDA Group Install/Delete failed");
-   }
-
-}
+           } else Fail.fail("Failed to intsall / delete resource");
+
+     } catch (CedaException | CemtException e) {
+        e.printStackTrace();
+     }
+  }
+
+  @Test
+  public void testResourceLibrary() throws TextNotFoundException, ErrorTextFoundException, Zos3270Exception, InterruptedException, CicstsManagerException {
+     String resourceType = "LIBRARY";
+     String resourceName = "lib1";
+     String groupName = "Test";
+     String resourceParameters = "DSNAME01(CTS.USER.APPL1.CICS.LOAD)";
+     boolean response = false;
+     try {
+        // Testing create and install resource by creating it, installing it and then checking if it appeared on CEMT
+        assertThat(cics.ceda().resourceExists(cedaTerminal, resourceType, resourceName, groupName)).isEqualTo(false);
+
+        cics.ceda().createResource(cedaTerminal, resourceType, resourceName, groupName, resourceParameters);
+        assertThat(cics.ceda().resourceExists(cedaTerminal, resourceType, resourceName, groupName)).isEqualTo(true);
+
+        cics.ceda().installResource(cedaTerminal, resourceType, resourceName, groupName);
+        if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) != null) {
+           response = true;
+        }
+        assertThat(response).isEqualTo(true);

```

Comment:
would it be simpler to assert that the returned hash map is not null and not empty?